core/src/main/java/com/linecorp/armeria/common/logging/ByteBufAggreatingPreviewer.java

```diff
@@ -32,39 +32,37 @@
 import io.netty.buffer.Unpooled;
 import io.netty.util.ReferenceCountUtil;
 
-abstract class ByteBufAggreatedPreviewer implements ContentPreviewer {
+abstract class ByteBufAggreatingPreviewer implements ContentPreviewer {
 
     private static final ByteBuf[] BYTE_BUFS = new ByteBuf[0];
-    private int capacity;
+    private int maxAggregatedLength;
     private final List<ByteBuf> bufferList;
     @Nullable
     private HttpHeaders headers;
     @Nullable
     private String produced;
     private int aggregatedLength;
 
-    ByteBufAggreatedPreviewer(int capacity) {
-        checkArgument(capacity >= 0, "capacity: %d (expected: >= 0)", capacity);
-        this.capacity = capacity;
+    ByteBufAggreatingPreviewer(int maxAggregatedLength) {
+        checkArgument(maxAggregatedLength >= 0,
+                      "maxAggregatedLength: %d (expected: >= 0)", maxAggregatedLength);
+        this.maxAggregatedLength = maxAggregatedLength;
         bufferList = new ArrayList<>();
     }
 
-    ByteBufAggreatedPreviewer() {
-        bufferList = new ArrayList<>();
-        capacity = 0;
+    ByteBufAggreatingPreviewer() {
+        this(0);
     }
 
-    void increaseCapacity(int delta) {
-        checkArgument(delta >= 0, "delta: %d (expected: >= 0)", delta);
-        if (Integer.MAX_VALUE - delta >= capacity) {
-            capacity = Integer.MAX_VALUE;
-        } else {
-            capacity += delta;
-        }
+    void maxAggregatedLength(int length) {
+        checkArgument(length > 0,
+                      "length: %d (expected: > 0", length);
+        maxAggregatedLength = length;
```

Comment:
How about adding an assertion: `assert maxAggregatedLength == 0 : "maxAggregatedLength cannot be set more than once: " + maxAggregatedLength;`